File: 240913/palindrome 여부 판단하기/determine-whether-palindrome-is-present.cpp
#include <iostream>
#include <vector>
using namespace std;

bool isPalindrome(vector<char>& a){
    vector<char> reverse;
    for (int i = a.size() - 1; i >= 0; i--){
        reverse.push_back(a[i]);
    }
    for (int i = 0; i < a.size(); i++){
        if(a[i] != reverse[i]){
            return false;
        }
    }
    return true;
}


int main() {
    //입력
    vector<char> a;
    string s;
    cin >> s;
    
    for (char c : s){
        a.push_back(c);
    }

    //연산 & 출력
    if (isPalindrome(a)){
        cout << "Yes";
    }
    else{
        cout << "No";
    }

    return 0;
}

File: 240913/palindrome 여부 판단하기/determine-whether-palindrome-is-present.java
import java.util.Scanner;

public class Main {

    public static boolean isPalindrome(String a){
        for (int i = 0; i < a.length(); i++){
            if(a.charAt(i) != a.charAt(a.length() - i - 1)){
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        //입력
        String a;
        Scanner sc = new Scanner(System.in);
        a = sc.nextLine();

        if(isPalindrome(a)){
            System.out.println("Yes");
        }
        else{
            System.out.println("No");
        }
    }
}